### PR TITLE
Upgrade to Rails 4

### PR DESCRIPTION
**Pull request by chadfennell**
* Test suite is green on Rails 4
* Fixed a couple of regressions encountered along the way
* Merged upstream FB auth and other changes
* Strong params Migration is still outstanding

It would be nice to have end-to-end functional tests for our primary happy user paths.

### CODE REVIEW

**Comment by pete2786** [comment]
Holy epic PR

File: app/models/organization.rb [hunk 1]
```diff
@@ -14,6 +13,14 @@ class Organization < ActiveRecord::Base
 
   scope :public_accessible, ->{ where(public_access: true) }
 
+  def self.current_id=(id)
```
Comment (pete2786) [inline app/models/organization.rb]: Nice

File: app/helpers/comment_mailer_helper.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+module CommentMailerHelper
+  def self.get_project_users(project)
+    ( project.project_owner.send_notifications? ? [project.project_owner] : [] ) | User.project_notifiable(project.id).to_a
```
Comment (pete2786) [inline app/helpers/comment_mailer_helper.rb]: I refactored this a bit to use a scope and got rid of the local variable which is kind of a general goal of our methods at sport ngin. Local variables are a last resort

File: app/models/project.rb [hunk 3]
```diff
@@ -43,7 +43,7 @@ def nil_if_blank
   def voting_allowed?
     # As of today, cannot vote if project is in 'parking lot'
     if self.event.nil? then
-      return true
+      return false
```
Comment (pete2786) [inline app/models/project.rb]: I think I like leaving this as true despite the contrary comment above.

File: app/models/project_comment.rb [hunk 4]
```diff
@@ -15,4 +17,8 @@ class ProjectComment < ActiveRecord::Base
   def slack_message
     "#{user.name} has commented on #{project.title}: #{description}"
   end
+
+  def send_notification
+    CommentMailer.comment_posted((self)).deliver
```
Comment (pete2786) [inline app/models/project_comment.rb]: Double parens just a typo here?

**Comment by pete2786** [comment]
Looks good besides minor comment issues. Also, do you know what we need to do to get travis passing?

File: app/models/project_comment.rb (hunk 4 above)
Comment (chadfennell) [inline app/models/project_comment.rb]: Oh, yep. I'll remove 'em and repost.

File: app/helpers/comment_mailer_helper.rb (hunk 2 above)
Comment (chadfennell) [inline app/helpers/comment_mailer_helper.rb]: Yeah, much nicer.

**Comment by chadfennell** [comment]
Travis will need the config.yml file sorted. Not having Travis is poking me in the eye too. I'll get it working after this PR for sure.

**Comment by pete2786** [comment]
w00t, finally got this deployed

**Pull request merged**